server/swap/swap.go

```diff
@@ -1832,7 +1832,7 @@ func (s *Swapper) processRedeem(msg *msgjson.Message, params *msgjson.Redeem, st
 	}
 
 	// Credit the user for completing the swap, adjusting the user's score.
-	if actor.user != counterParty.user || newStatus == order.MatchComplete { // if user is both sides, only credit on MatchComplete (taker done too)
```

Comment:
Should the `||` have been an `&&`? It looks fine with just the not counterparty check. User should get not credit for trading with themself.

Reply:
I'm guessing the intention was to give someone one credit when trading with themselves instead of two.

Reply:
Actually, you're both right. Someone did intend to give 1 point for setf-matches, but we shouldn't.

Reply:
Do we need the `newStatus == order.MatchComplete`? Why should the maker have to wait for the taker to also redeem before getting their score?

Reply:
No? Looks ok to me.

Reply:
> Do we need the `newStatus == order.MatchComplete`? Why should the maker have to wait for the taker to also redeem before getting their score?

Oh right. No need for the `newStatus == order.MatchComplete`.